mantidproject/mantid
# Make sure axis labelling is correct in vsi

Fixes #13472 
Fixes #13209 which is a duplicate
### For Testing

Please find sample workspaces here: \olympic\Babylon5\Scratch\Anton\VSI\SAMPLE_WORKSPACES
Relevant files are:
- MDEvent_Osiris.nxs
- MDHisto_Osiris.nxs
- MAPS_MDEW
- SEQ_MDHW

**Testing the Axes Grid system**

Perform this test with MD_Event_Osiris and once with MDHisto_Osiris. The workspace is called WS from here on.
1. Load WS into the VSI
   - Confirm that the Axes Grid system is enabled
   - Confirm that the checkbox next to the button "Edit Axes Grid" is checked
   - Confirm that the units are rendered in Latex style inverse Angstroms
   - Confirm that the unit system is "Q_lab_x, Q_lab_y, Q_lab_z" 
2. Switch to Three-Slice-View
   - Confirm that all four views have the Axes Grid system. Note this requires the ParaView Build to be patched, if you only see the Axes Grid in the lower-right-hand corner, then this is not due to this ticket, but due to the missing patch
3. Switch to the Multislice View 
   - Confirm that the Axes Grid system is enabled
4. Switch to the Standard view and apply a Scale filter
   - Confirm that the Axes Grid is still enabled
5. Delete the Scale filter
6. Uncheck the Axes Grid system. ( The  checkbox next to the"Edit Axes Gird" button)
7. Switch to a different view
   - Confirm that the Axes Grid system is still disabled and not visible
8. Close the VSI

**Testing labels for specific data sets**

You need the following data sets:
- MAPS_MDEW
- SEQ_MDHW

We have to rebin MAPS_MDEW with the following script:

``` python
# First load the MAPS_MDEW.nxs file into Mantid
ws = mtd['MAPS_MDEW']
ws_binned = BinMD(InputWorkspace = ws, 
                          AlignedDim0 ='A,0,10,10', 
                          AlignedDim1 ='B,0,10,10', 
                          AlignedDim2 ='C,0,10,10', 
                          AlignedDim3 ='D,0,10,10')
```
1. Load the data rebinned MAPS data set into the VSI and go to Standar View
   - Confirm that the latex labels show an Angstrom symbol. Note there will be nothing visible (maybe bad rebinning?) and hence the Axes Grid looks slightly odd. This is a ParaView "issue" over which we don't have any influence.
2. Close the VSI
3. Load the data set called SEQ_MDHW into the VSI, got to Standard View
   - Confirm that the latex labels show somethign like ("in" someNumber  inverseAngstrom)
